pygbm/gradient_boosting.py

```diff
@@ -196,8 +214,56 @@ def _should_stop(self, scores):
         return candidate >= best_with_tol
 
 
-# TODO: shall we split between GBMClassifier and GBMRegressor instead
-# of using a single class?
+class GradientBoostingRegressor(BaseGradientBoostingMachine, RegressorMixin):
+
+    _SUPPORTED_LOSS = ('ls',)
+
+    def __init__(self, loss='ls', learning_rate=0.1, max_iter=100,
+                 max_leaf_nodes=31, max_depth=None, min_samples_leaf=20,
+                 l2_regularization=0., max_bins=256, max_no_improvement=5,
+                 validation_split=0.1, scoring='neg_mean_squared_error',
+                 tol=1e-7, verbose=0, random_state=None):
+        super(GradientBoostingRegressor, self).__init__(
+            loss=loss, learning_rate=learning_rate, max_iter=max_iter,
+            max_leaf_nodes=max_leaf_nodes, max_depth=max_depth,
+            min_samples_leaf=min_samples_leaf,
+            l2_regularization=l2_regularization, max_bins=max_bins,
+            max_no_improvement=max_no_improvement,
+            validation_split=validation_split, scoring=scoring, tol=tol,
+            verbose=verbose, random_state=random_state)
+
+    def predict(self, X):
+        return self._decision_function(X)
```

Comment:
It's a bit weird to call a method named `_decision_function` in a regressor. Maybe it should still be called `predict` in the base class so that it can be inherited directly in the regressor and a be overriden by a loss specific method in the classifier subclass. Maybe some regression loss function would also need to have the loss be able to post process the raw predictions of the trees.

Alternatively call it `_raw_predict` in the base class.

Reply:
That's what scikitlearn is doing but I agree, that was actually a question I had for you in #49.

I proposed `_get_scores()` but I like `_raw_predict()` as well